iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/node/WALNode.java

```diff
@@ -219,34 +222,75 @@ private class DeleteOutdatedFileTask implements Runnable {
     // .wal files whose version ids are less than first valid version id should be deleted
     private long firstValidVersionId;
     // the effective information ratio
-    private double effectiveInfoRatio;
-    // recursion time of calling deletion
+    private double effectiveInfoRatio = 0d;
+
+    private List<Long> pinnedMemTableIds;
+
+    private File[] filesShouldDelete;
+
+    private int fileIndexAfterFilterSafelyDeleteIndex = Integer.MAX_VALUE;
```

Comment:
set this to Integer.MAX_VALUE in the init() method

Reply:
This problem can be ignored after confirmation